### PR TITLE
powershell based install script

### LINKED ISSUE #5437: install.ps1 (install.sh equivalent) needed
OS:  All supported windows distributions

In order to streamline the habitat node bootstrapping process across platforms there should be an install.ps1 script similar to the install.sh (https://github.com/habitat-sh/habitat/blob/master/components/hab/install.sh) script for windows installation.  This is also necessary for platform agnostic windows capability in the habitat cookbook (imho).

### PR DESCRIPTION
**Pull request by mwrock**
Fixes #5437 

This should work for all supported versions of windows (2008 R2 forward):

```
Set-ExecutionPolicy Bypass -Scope Process -Force
iex ((New-Object System.Net.WebClient).DownloadString('https://raw.githubusercontent.com/habitat-sh/habitat/install_ps1/components/hab/install.ps1'))
```

Note that this also adjusts our CI to install hab from this script instead of chocolatey.

Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

File: www/source/partials/global/_download-and-install.slim [hunk 1]
```diff
@@ -43,6 +43,13 @@ hr
       = code(:powershell) do
         | $ choco install habitat
 
+      h4.mt-2 Install Chef Habitat using a Powershell instal script
```
Comment (raskchanky) [inline www/source/partials/global/_download-and-install.slim]: `instal` -> `install`

File: components/hab/install.ps1 [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            $e += " Please provide a simple version like: '0.15.0'"
+            $e += " or a fully qualified version like: '0.15.0/20161222203215'."
+            Write-Error $e
+        } else {
+            Write-Host "Using fully qualified Bintray version string of: $rev"
+            $version = $rev
+        }
+    } else {
+        # No ConvertFrom-Json support so we must be on unpatched PSv2
+        if($version -match "^\d\.\d\d\.\d-\d\d\d\d\d\d\d\d\d\d\d\d\d\d`$") {
```
Comment (christophermaier) [inline components/hab/install.ps1]: Is this matching a string like `0.75.0/20190219232208`?

If so, this is probably too restrictive... nothing guarantees that our major version will only ever be one digit, or our minor version two digits and our patch one digit (in fact, as we move to "release with each merge", it's going to be fairly common to see triple-digit minor and patch versions).

Does Powershell regex have syntax for quantifying numbers of repetitions? Something like `\d+\.\d+\.\d+/\d{12}` would be much more readable, for instance.

Also, is this intending to match only something like `0.75.0/20190219232208`, or should it also try to match `0.75.0`, too?

File: README.md [hunk 3]
```diff
@@ -85,13 +85,19 @@ If you are running Windows and use [Chocolatey](https://chocolatey.org), you can
...
 
+Bash:
 ```
 $ curl https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh | sudo bash
 ```
 
+Powershell:
+```
+C:\> Set-ExecutionPolicy Bypass -Scope Process -Force
+C:\> iex ((New-Object System.Net.WebClient).DownloadString('https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.ps1'))
```
[the remark this answers is not in the document]
Comment (mwrock) [reply README.md]: there are somewhat better options like `. { iwr https://raw.githubusercontent.com/habitat-sh/habitat/install_ps1/components/hab/install.ps1 } | iex` but that won't work on older versions of PS (but most). I'm trying to capture our entire support base.

File: components/hab/install.ps1 (hunk 2 above)
Comment (mwrock) [reply components/hab/install.ps1]: ah good catch. your `\d+\.\d+\.\d+/\d{12}` works. Thanks!

Comment (mwrock) [reply components/hab/install.ps1]: the intent is to make sure it is fully qualified. If we cant parse the json file to find the actual release we are trusting the user is giving it to us.

Comment (baumanj) [reply components/hab/install.ps1]: That `12`s gonna break in 7,981 years

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (christophermaier) [reply README.md]: @mwrock I was totally being snarky, but I appreciate your explanation, as well as the thought given to users of older powershell installations ❤️

File: components/hab/install.ps1 (hunk 2 above)
Comment (christophermaier) [reply components/hab/install.ps1]: The oft-overlooked Y10K problem...

**Pull request merged**